kernel/aster-nix/src/net/iface/common.rs

```diff
@@ -164,7 +164,9 @@ impl IfaceCommon {
             let mut has_events = false;
             let mut poll_at;
             loop {
-                has_events |= interface.poll(timestamp, device, &mut sockets);
+                while interface.poll(timestamp, device, &mut sockets) {
+                    has_events = true;
+                }
```

Comment:
This looks a bit strange. The outer and inner loops have some overlapping semantics.

Could you please explain more about why you made this change and why the outer loop cannot solve the problem?

Reply:
Yes, I made this change because the CI gets stuck when running the udp test (`test/apps/network/udp_client.`). I found the cause of the issue is this [`commit in smoltcp`](https://github.com/smoltcp-rs/smoltcp/commit/8c2cef19965acda8f7d602a2f36d0f81f55ef54e). This commit changes the loop inside the poll to a single iteration, causing the udp client to fail to receive the message from the server.

Reply:
Thanks for your explanation! This make senses.

Then I'm curious if the outer loop is really necessary, since the inner loop should ensure that all pending packets are sent or received. But that can be considered a separate issue.

At the very least, I suggest adding some comments describing the reason for introducing each loop.

```rust
            loop {
                // `poll` transmits and receives a bounded number of packets. This loop ensures
                // that all packets are transmitted and received. For details, see
                // <https://github.com/smoltcp-rs/smoltcp/blob/8e3ea5c7f09a76f0a4988fda20cadc74eacdc0d8/src/iface/interface/mod.rs#L400-L405>.
                while interface.poll(timestamp, device, &mut sockets) {
                    has_events = true;
                }

                // `poll_at` can return `Some(Instant::from_millis(0))`, which means `PollAt::Now`.
                // For details, see
                // <https://github.com/smoltcp-rs/smoltcp/blob/8e3ea5c7f09a76f0a4988fda20cadc74eacdc0d8/src/iface/interface/mod.rs#L478>.
                poll_at = interface.poll_at(timestamp, &sockets);
                let Some(instant) = poll_at else {
                    break;
                };
                if instant > timestamp {
                    break;
                }
            }
```